libvast/include/vast/pipeline.hpp

```diff
@@ -174,11 +174,16 @@ class pipeline final : public operator_base {
   /// pipelines, for example `(a | b) | c` becomes `a | b | c`.
   explicit pipeline(std::vector<operator_ptr> operators);
 
-  /// Parses a logical pipeline from its textual representation. It is *not*
-  /// guaranteed that `parse(to_string())` succeeds.
+  /// Parses a pipeline from its textual representation.
   static auto parse(std::string_view repr, const vast::record& config)
     -> caf::expected<pipeline>;
```

Comment:
If we want to require the round-trip property, I would mention it in the documentation.

Reply:
I like that. I'll add a variant of this even though we don't have `operator_location` in this branch yet.